src/EPROMStore.cpp

```diff
@@ -4,152 +4,659 @@
 
 // The platform-independant EEPROM class
 
+///////////////////////////////////////
+// PLATFORM-SPECIFIC IMPLEMENTATIONS
+
 #if USE_DUMMY_EEPROM == 1
 
-static uint8_t dummyEepromStorage[EPROMStore::STORE_SIZE];
+static uint8_t dummyEepromStorage[EEPROMStore::STORE_SIZE];
 
 // Initialize the EEPROM object for ESP boards, setting aside storage
-void EPROMStore::initialize()
+void EEPROMStore::initialize()
 {
-  LOGV2(DEBUG_EEPROM, F("EEPROM[DUMMY]: Startup with %d bytes"), EPROMStore::STORE_SIZE);
+  LOGV2(DEBUG_EEPROM, F("EEPROM[DUMMY]: Startup with %d bytes"), EEPROMStore::STORE_SIZE);
   memset(dummyEepromStorage, 0, sizeof(dummyEepromStorage));
 }
 
 // Update the given location with the given value
-void EPROMStore::update(uint8_t location, uint8_t value)
+void EEPROMStore::update(uint8_t location, uint8_t value)
 {
-  LOGV3(DEBUG_EEPROM, F("EEPROM[DUMMY]: Writing %x to %d"), value, location);
+  LOGV3(DEBUG_EEPROM, F("EEPROM[DUMMY]: Writing %02X to %d"), value, location);
   dummyEepromStorage[location] = value;
 }
 
+// Complete the transaction
+void EEPROMStore::commit()
+{
+  // Nothing to do
+}
+
 // Read the value at the given location
-uint8_t EPROMStore::read(uint8_t location)
+uint8_t EEPROMStore::read(uint8_t location)
 {
   uint8_t value;
   value = dummyEepromStorage[location];
-  LOGV3(DEBUG_EEPROM, F("EEPROM[DUMMY]: Read %x from %d"), value, location);
+  LOGV3(DEBUG_EEPROM, F("EEPROM[DUMMY]: Read %02X from %d"), value, location);
   return value;
 }
 
 #elif defined(ESPBOARD)
```

Comment:
We could do. It's not too bad right now as the differences are very small and clear. Its a trade off between proliferation of files and clarity of implementation. The code in EEPROM is so boring its probably not worth it right now - the axis controllers are perhaps a different story.

Reply:
mh fair enough :)

Reply:
You cannot change %x to %02X. The LOGVx family uses our own formatter and it does not support the full printf() format spec. I think I had to implement this because one of the platforms didn't support sprintf(). If you can get sprintf() to work on our platforms, we could replace formatArg() in Utility.cpp with a sprintf() call.

Reply:
Done. So much for the last minute cosmetic change :-(